kernel/src/engine/ensure_data_types.rs

```diff
@@ -44,6 +44,7 @@ struct EnsureDataTypes {
 }
 
 /// Capture the compatibility between two data-types, as passed to [`ensure_data_types`]
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
Why do we need debug and clone? It does have a compile time cost, so I prefer to add it only when needed. Also, if the eq/partialeq are just for tests, can we only derive it if testing? (not sure how easy/possible that is)